webapp/packages/core-resource/src/Resource/OffsetPagination/ResourceOffsetPage.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { makeObservable, observable } from 'mobx';
+
+import { IResourceOffsetPage } from './IResourceOffsetPage';
+
+export class ResourceOffsetPage implements IResourceOffsetPage {
+  from: number;
+  to: number;
+  items: any[];
+  outdated: boolean;
+
+  constructor() {
+    this.from = 0;
+    this.to = 0;
+    this.items = [];
+    this.outdated = false;
+
+    makeObservable(this, {
+      from: true,
+      to: true,
+      items: observable.shallow,
+      outdated: observable,
```

Comment:
observable.ref maybe? as its a boolean

Reply:
there is no difference for primitives actually `observable` and `observable.ref` will have same effect